libcxx/include/__availability

```diff
@@ -299,4 +310,11 @@
 #  define _LIBCPP_AVAILABILITY_THROW_BAD_VARIANT_ACCESS _LIBCPP_AVAILABILITY_BAD_VARIANT_ACCESS
 #endif
 
+// Define availability attributes that depend on both
+// _LIBCPP_HAS_NO_EXCEPTIONS and _LIBCPP_HAS_NO_RTTI.
+#if defined(_LIBCPP_HAS_NO_EXCEPTIONS) || defined(_LIBCPP_HAS_NO_RTTI)
+#   undef _LIBCPP_AVAILABILITY_HAS_INIT_PRIMARY_EXCEPTION
```

Comment:
You also need to `#undef _LIBCPP_AVAILABILITY_INIT_PRIMARY_EXCEPTION` and `#define _LIBCPP_AVAILABILITY_INIT_PRIMARY_EXCEPTION` to make that macro a no-op when RTTI / exceptions are disabled.